### PR TITLE
First part of test refactoring to improve reporting

### PR DESCRIPTION
**Pull request by kevinearls**
Signed-off-by: Kevin Earls <kearls@redhat.com>

This PR is the first part of #385 and covers the smoke tests. It includes restructuring tests and using the stretchr/testify toolkit to help produce better test reporting.

### CODE REVIEW

**Comment by kevinearls** [comment]
@pavolloffay  @objectiser  @jpkrohling  I've resubmitted this as a new PR because of a small git related catastrophe.  :-). @objectiser  I think I have covered all or your comments from the first submission, but please let me know if I've missed anything.

File: test/e2e/all_in_one_test.go [hunk 1]
```diff
@@ -153,78 +133,34 @@ func allInOneWithIngressTest(t *testing.T, f *framework.Framework, ctx *framewor
...
-		return err
-	}
-
-	err = e2eutil.WaitForDeployment(t, f.KubeClient, namespace, "all-in-one-with-ui-config", 1, retryInterval, timeout)
-	if err != nil {
-		return err
-	}
+	j := getJaegerAllInOneWithUiDefinition(basePath)
+	err := fw.Client.Create(goctx.TODO(), j, cleanupOptions)
+	require.NoError(t, err, "Error creating jaeger-operator instance")
```
Comment (objectiser) [inline test/e2e/all_in_one_test.go]: Should be "jaeger instance" - the operator is deployed by the operator-sdk test framework.

File: test/e2e/all_in_one_test.go [hunk 2]
```diff
@@ -153,78 +133,34 @@ func allInOneWithIngressTest(t *testing.T, f *framework.Framework, ctx *framewor
...
+	err = e2eutil.WaitForDeployment(t, fw.KubeClient, namespace, "all-in-one-with-ui-config", 1, retryInterval, timeout)
+	require.NoError(t, err, "Error waiting for operator deployment")
```
Comment (objectiser) [inline test/e2e/all_in_one_test.go]: Should be "waiting for jaeger deployment" - operator has already been deployed by the test framework.

File: test/e2e/utils.go [hunk 3]
```diff
@@ -22,6 +28,10 @@ var (
 	storageNamespace     = os.Getenv("STORAGE_NAMESPACE")
 	esServerUrls         = "http://elasticsearch." + storageNamespace + ".svc:9200"
 	cassandraServiceName = "cassandra." + storageNamespace + ".svc"
+    ctx 					*framework.TestCtx
```
Comment (objectiser) [inline test/e2e/utils.go]: Indentation issues - `make format` should hopefully sort it out.

File: test/e2e/utils.go [hunk 4]
```diff
@@ -28,10 +28,10 @@ var (
...
 	esServerUrls         = "http://elasticsearch." + storageNamespace + ".svc:9200"
 	cassandraServiceName = "cassandra." + storageNamespace + ".svc"
-    ctx 					*framework.TestCtx
-    fw 						*framework.Framework
-    namespace 				string
-    t 						*testing.T
+	ctx 					*framework.TestCtx
+	fw 						*framework.Framework
+	namespace 				string
+	t 						*testing.T
```
Comment (objectiser) [inline test/e2e/utils.go]: Was this after running `make format`? Seems strange, unless it is having problems as these vars don't have `=`

Comment (kevinearls) [reply test/e2e/utils.go]: @objectiser Yes, it was after make format.  Sorry I didn't check before I submitted it, but I think I've gotten it straightened out now.

**Pull request merged**